Resources/Prototypes/floor_trap.yml

```diff
@@ -0,0 +1,196 @@
+- type: entity
+  id: CollideFloorTrap
+  abstract: true
+  placement:
+    mode: SnapgridCenter
+  components:
+    - type: Sprite
+      sprite: Tiles/Misc/floortrap.rsi
+      state: floortrap
+    - type: Fixtures
+      fixtures:
+        floortrap:
+          shape:
+            !type:PhysShapeAabb
+            bounds: "-0.4,-0.4,0.4,0.4"
+          hard: false
+          mask:
+            - ItemMask
+          layer:
+            - SlipLayer
+    - type: Physics
+    - type: Tag
+      tags:
+        - HideContextMenu
+
+- type: entity
+  parent: CollideFloorTrap
+  id: CollideFloorTrapSpawn
+  name: floor trap spawn
+  abstract: true
+  components:
+    - type: Sprite
+      sprite: Tiles/Misc/floortrap.rsi
+      state: floortrapspawn
+
+- type: entity
+  id: MobSpawnFloorTrap
+  name: mobspawner floor trap
+  abstract: true
+  components:
+    - type: Transform
+      anchored: True
+    - type: InteractionOutline
+    - type: Physics
+      bodyType: Static
+    - type: Sprite
+      sprite: /Textures/Effects/teleport.rsi
+      state: teleport
+    - type: Fixtures
+      fixtures:
+        portalFixture:
+          shape:
+            !type:PhysShapeAabb
+            bounds: "-0.25,-0.48,0.25,0.48"
+          mask:
+          - FullTileMask
+          layer:
+          - WallLayer
+          hard: false
+
+- type: entity
+  parent: CollideFloorTrap
+  id: FloorTrapExplosion
+  name: explosion floor trap
+  components:
+    - type: TriggerOnCollide
+      fixtureID: floortrap
+    - type: ExplodeOnTrigger
+    - type: Explosive
+      explosionType: Default
+      totalIntensity: 20.0
+      intensitySlope: 5
+      maxIntensity: 4
+    - type: DeleteOnTrigger
+
+- type: entity
+  parent: CollideFloorTrap
+  id: FloorTrapEMP
+  name: EMP floor trap
+  components:
+    - type: TriggerOnCollide
+      fixtureID: floortrap
+    - type: EmpOnTrigger
+      range: 2
+      energyConsumption: 5000
+    - type: DeleteOnTrigger
+
+- type: entity
+  parent: CollideFloorTrapSpawn
+  id: SpawnFloorTrapCarp
+  name: spawn floor trap
+  suffix: Carp
+  components:
+    - type: TriggerOnCollide
+      fixtureID: floortrap
+    - type: SpawnOnTrigger
+      proto: MobSpawnFloorTrapCarp
+    - type: DeleteOnTrigger
```

Comment:
Why doesn't it just spawn the carp directly? Seems like a waste of an entire entity to just relay this.

Reply:
This is done for several reasons:
1. It's prettier.
2. it doesn't happen instantly, and the player has time to realise and react to the appearance of the mob. For this reason, I didn't add the space spider and cobra spawn, which have poisons and can be extremely dangerous by suddenly appearing near the player.
3. The effect itself gives the understanding that the player has entered the trap, it is an indicator of the event. Than if a mobe suddenly appeared near the player, which is more like adminspawn.

Reply:
If you want a timer implement a timer into it, don't spawn entities just to work around it.